src/integration-test/java/com/commercetools/sync/integration/services/impl/CartDiscountServiceImplIT.java

```diff
@@ -0,0 +1,432 @@
+        assertThat(cachedCartDiscountId).contains(cartDiscountOptional.get().getId());
+        assertThat(errorCallBackExceptions).isEmpty();
+        assertThat(errorCallBackMessages).isEmpty();
+    }
+
+    @Test
+    void createCartDiscount_WithValidCartDiscount_ShouldCreateCartDiscountAndCacheId() {
+        //preparation
+        final CartDiscountDraft newCartDiscountDraft =
+                CartDiscountDraftBuilder.of(LocalizedString.of(Locale.ENGLISH, CART_DISCOUNT_KEY_1),
+                        CART_DISCOUNT_CART_PREDICATE_1,
+                        CART_DISCOUNT_VALUE_1,
+                        CART_DISCOUNT_TARGET_1,
+                        SORT_ORDER_2,
+                        false)
+                        .active(false)
+                        .build();
+
+        final SphereClient spyClient = spy(CTP_TARGET_CLIENT);
+
+        final CartDiscountSyncOptions spyOptions =
+                CartDiscountSyncOptionsBuilder.of(spyClient)
+                        .errorCallback((errorMessage, exception) -> {
+                            errorCallBackMessages.add(errorMessage);
+                            errorCallBackExceptions.add(exception);
+                        })
+                        .build();
+
+        final CartDiscountService spyCartDiscountService = new CartDiscountServiceImpl(spyOptions);
+
+        //test
+        final Optional<CartDiscount> createdCartDiscount = spyCartDiscountService
+                .createCartDiscount(newCartDiscountDraft)
+                .toCompletableFuture().join();
+
+        final Optional<CartDiscount> queriedOptional = CTP_TARGET_CLIENT
+                .execute(CartDiscountQuery.of().withPredicates(cartDiscountQueryModel ->
+                        cartDiscountQueryModel.name().lang(Locale.ENGLISH).is(CART_DISCOUNT_KEY_1)))
+                .toCompletableFuture().join().head();
+
+        assertThat(queriedOptional).hasValueSatisfying(queried ->
+                assertThat(createdCartDiscount).hasValueSatisfying(created -> {
+                    assertThat(created.getName()).isEqualTo(queried.getName());
+                    assertThat(created.getCartPredicate()).isEqualTo(queried.getCartPredicate());
+                    assertThat(created.getValue()).isEqualTo(queried.getValue());
+                    assertThat(created.getTarget()).isEqualTo(queried.getTarget());
+                    assertThat(created.getSortOrder()).isNotEqualTo(queried.getSortOrder());
+                }));
+
+        // Assert that the created cart discount is cached
+        final Optional<String> cartDiscountId =
+                spyCartDiscountService.fetchCachedCartDiscountId(CART_DISCOUNT_KEY_1).toCompletableFuture().join();
+        assertThat(cartDiscountId).isPresent();
+        verify(spyClient, times(0)).execute(any(CartDiscountQuery.class));
+    }
+
+    @Test
+    void createCartDiscount_WithInvalidCartDiscount_ShouldHaveEmptyOptionalAsAResult() {
```

Comment:
a unit test for this is better, since no is call is needed and no mocking is needed.

Reply:
f7a82da